src/lib/defaultMap.js

```diff
@@ -0,0 +1,24 @@
+ /**
+ * Copyright (c) 2016-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+'use strict';
+
+module.exports = function defaultMap(factory, iterable) {
```

Comment:
`MapWithDefaultFactory`? `extends Map` is a no-no, I tried it. It doesn’t like the `Map.call(this)` that babel creates. Maybe it would work if we’d switch the transform to something like https://www.npmjs.com/package/babel-preset-es2015-node4 or https://www.npmjs.com/package/babel-preset-modern-node